pkg/publisher/publisher.go

```diff
@@ -11,11 +11,30 @@ import (
 
 var log = logrus.New()
 
-func Publish(ip net.IP, iface net.Interface, service Service, shutdown chan struct{}, waitGroup *sync.WaitGroup) (err error) {
+func Publish(iface net.Interface, service Service, shutdown chan struct{}, waitGroup *sync.WaitGroup) (err error) {
 	defer waitGroup.Done()
+	var v4addr, v6addr net.IP
+	addrs, err := iface.Addrs()
```

Comment:
I don't think we can do this. The reason the IP was specified before was that if the VIP is present on a given system it will also show up in this list and we don't want to publish that as our address.

Reply:
> I don't think we can do this. The reason the IP was specified before was that if the VIP is present on a given system it will also show up in this list and we don't want to publish that as our address.

Ok... So I guess the question is whether we want to expose IPv4 and IPv6 at the same time? That's the reason for looping on the addrs... I'll have to rethink this.